### PR TITLE
Add Linux version code macro for ch2/procfs_simple_intf example for newer kernels

### PR DESCRIPTION
**Pull request by seantywork**
- file_operations not supported thus added proc_ops
 - kzfree() not supported thus added kfree_sensitive()
 
* warn: 6,5,0 version for kzfree -> kfree_sensitive is somewhat arbitrary as which version exactly that change came in is somewhat blurry to me. So I put in my kernel version against which the code has been tested

### CODE REVIEW

File: ch2/procfs_simple_intf/procfs_simple_intf.c [hunk 1]
```diff
@@ -384,7 +411,12 @@ static int __init procfs_simple_intf_init(void)
 	return 0;	/* success */
 
  out_fail_3:
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 5, 0)
```
Comment (kaiwan) [inline ch2/procfs_simple_intf/procfs_simple_intf.c:415]: it's from ver 5.9.0 that kfree_sensitive() replaces kzfree(); pl update, thanks

File: ch2/procfs_simple_intf/procfs_simple_intf.c [hunk 2]
```diff
@@ -395,7 +427,11 @@ static void __exit procfs_simple_intf_cleanup(void)
 {
 	gdrvctx->power = 0;
 	remove_proc_subtree(OURMODNAME, NULL);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 5, 0)
```
Comment (kaiwan) [inline ch2/procfs_simple_intf/procfs_simple_intf.c:430]: ditto to above comment

**Review by kaiwan — CHANGES_REQUESTED** [review]

File: ch2/procfs_simple_intf/procfs_simple_intf.c (hunk 1 above)
Comment (seantywork) [reply ch2/procfs_simple_intf/procfs_simple_intf.c]: Updated. Thank you very much again for another piece of knowledge!

File: ch2/procfs_simple_intf/procfs_simple_intf.c (hunk 2 above)
Comment (seantywork) [reply ch2/procfs_simple_intf/procfs_simple_intf.c]: Again thanks for the help!

Comment (kaiwan) [reply ch2/procfs_simple_intf/procfs_simple_intf.c]: Welcome! Thanks @seantywork for the work, these updates!

**Pull request merged**